packages/message-system/package.json

```diff
@@ -25,5 +25,20 @@
         "jest": "^26.6.3",
         "tsx": "^3.8.2",
         "typescript": "4.7.4"
+    },
+    "nx": {
+        "targets": {
+            "build:lib": {
+                "outputs": [
+                    "./src/types/messageSystem.ts"
+                ]
+            },
+            "type-check": {
+                "dependsOn": [
+                    "^build:lib",
+                    "build:lib"
```

Comment:
`^build:lib` this means that that it will run `build:lib` in all dependencies of this package before it will run `type-check` here. This is actually set also globally, but because I needed to add that line lower
`build:lib` (without `^`) - mean it will run `build-lib` in this package before running `type-check`